components/usage/pkg/db/workspace_instance.go

```diff
@@ -71,16 +56,20 @@ func (i *WorkspaceInstance) TableName() string {
 // - running
 // - instances which only just terminated after the start period
 // - instances which only just started in the period specified
-func ListWorkspaceInstancesInRange(ctx context.Context, conn *gorm.DB, from, to time.Time) ([]WorkspaceInstance, error) {
-	var instances []WorkspaceInstance
-	var instancesInBatch []WorkspaceInstance
+func ListWorkspaceInstancesInRange(ctx context.Context, conn *gorm.DB, from, to time.Time) ([]WorkspaceInstanceForUsage, error) {
+	var instances []WorkspaceInstanceForUsage
+	var instancesInBatch []WorkspaceInstanceForUsage
+
 	tx := conn.WithContext(ctx).
+		Table(fmt.Sprintf("%s as wsi", (&WorkspaceInstance{}).TableName())).
+		Select("wsi.id as id, ws.projectId as projectId, ws.type as workspaceType, wsi.workspaceClass as workspaceClass, wsi.usageAttributionId as usageAttributionId, wsi.stoppedTime as stoppedTime, wsi.creationTime as creationTime").
+		Joins(fmt.Sprintf("LEFT JOIN %s AS ws ON wsi.workspaceId = ws.id", (&Workspace{}).TableName())).
 		Where(
-			conn.Where("stoppedTime >= ?", TimeToISO8601(from)).Or("stoppedTime = ?", ""),
+			conn.Where("wsi.stoppedTime >= ?", TimeToISO8601(from)).Or("wsi.stoppedTime = ?", ""),
 		).
-		Where("creationTime < ?", TimeToISO8601(to)).
-		Where("startedTime != ?", "").
-		Where("usageAttributionId != ?", "").
+		Where("wsi.creationTime < ?", TimeToISO8601(to)).
+		Where("wsi.startedTime != ?", "").
+		Where("wsi.usageAttributionId != ?", "").
```

Comment:
Do we need to test this query against larger data sets to see how it performs, or happy to deploy to prod as is?

Reply:
I can give it a go against prod data